pkg/solidity-utils/contracts/helpers/BufferHelpers.sol

```diff
@@ -53,7 +56,10 @@ library BufferHelpers {
 
         uint256 underlyingBalanceAsWrapped = 0;
         if (bufferBalance.getBalanceRaw() > 0) {
-            underlyingBalanceAsWrapped = wrappedToken.convertToShares(bufferBalance.getBalanceRaw());
+            // Buffer wrapped surplus is used when unwrapping (it means, deposit wrapped and get underlying tokens),
+            // so we use `previewWithdraw` to convert underlying balance to wrapped. `withdraw` is the exact opposite
+            // operation as `redeem`, used to unwrap wrapped tokens.
+            underlyingBalanceAsWrapped = wrappedToken.previewWithdraw(bufferBalance.getBalanceRaw());
```

Comment:
Consider replacing "`withdraw` is the exact opposite operation as `redeem`, used to unwrap wrapped tokens."
with: "The withdraw function is used here, as it performs the inverse of a redeem operation." for consistency with the comments below.